### PR TITLE
add Hebrew translation

### PR DESCRIPTION
**Pull request by Ohohko**
#### What kind of changes does this PR include?

- Minor content fixes (broken links, typos, etc.)
- Changes or translations of Starlight docs site content
- Changes to Starlight code
- Something else!

#### Description

- Closes # 
- What does this PR change? Give us a brief description.
- Did you change something visual? A before/after screenshot can be helpful.

### CODE REVIEW

File: packages/starlight/translations/he.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
+  {
+    "skipLink.label": "דלגו לתוכן",
+    "search.label": "חיפוש",
+    "search.shortcutLabel": "(לחצו / לחיפוש)",
+    "search.cancelLabel": "ביטול",
+    "search.devWarning": "לא ניתן להציג תצוגה מקדימה של האתר באופן מקומי.",
```
Comment (delucis) [inline packages/starlight/translations/he.json:6]: The English version here says a bit more than this sentence:

```
Search is only available in production builds. \nTry building and previewing the site to test it out locally.
```

Could you update to be a bit closer so it specifies the option of building and previewing as well?

File: packages/starlight/translations/he.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "themeSelect.accessibleLabel": "בחרו פרופיל צבע",
+    "themeSelect.dark": "כהה",
+    "themeSelect.light": "בהיר",
+    "themeSelect.auto": "מערכת",
+    "languageSelect.accessibleLabel": "בחרו שפה",
+    "menuButton.accessibleLabel": "תפריט",
+    "sidebarNav.accessibleLabel": "ראשי",
+    "tableOfContents.onThisPage": "בדף זה",
+    "tableOfContents.overview": "סקירה כללית",
+    "i18n.untranslatedContent": "תוכן זה אינו זמין עדיין בשפת העברית.",
```
Comment (delucis) [inline packages/starlight/translations/he.json:16]: Might be Google Translate tripping me up, but testing there, this translates as “This content is not yet available **in Hebrew**.” but in English we use “This content is not available in **your language** yet.”

This is a subtle difference but it’s hypothetically possible depending on site set up that this notice might be displayed in Hebrew to someone who wants content in a different language, so should be translated more generically as “your language”.

**Review by delucis — COMMENTED** [review]
Thank you for the PR @Ohohko! This looks great. I just spotted a couple of details to check.

One thing: can we add the 404.md in a second PR instead of as part of this one?

That way this PR can be specifically for adding stuff to `packages/starlight/`, which is the code we publish to be used by all Starlight users, and then can use a future PR to set up Hebrew on the Starlight docs site itself.

File: packages/starlight/translations/he.json (hunk 2 above)
Comment (Ohohko) [reply packages/starlight/translations/he.json:16]: done

**Comment by delucis** [comment]
Oh, noticed you closed this @Ohohko! Anything I can help out with?

**Pull request closed without merging**